### PR TITLE
cache: examples of DI

### PR DESCRIPTION
**Pull request by milo**

### CODE REVIEW

File: cs/caching.texy [hunk 1]
```diff
@@ -303,5 +303,55 @@ $result = $cache->save($key, function() { // nebo callback(...)
...
+	private $cache;
+
+	public function __construct(Caching\IStorage $storage)
+	{
+		$this->cache = new Caching\Cache($storage, 'my-service');
+	}
+}
+\--
+
+Služba dostane instaci úložiště z DI kontejneru:
```
Comment (Majkl578) [inline cs/caching.texy]: instanci

File: en/caching.texy [hunk 2]
```diff
@@ -305,5 +305,55 @@ Framework will ensure that the body of the function will be called only by one t
 If the thread fails for some reason, another gets chance.
 
 
+Using cache accross the application
+===================================
+
+When using cache in services, we are facing to decision, if to pass a `Cache` or `IStoarge` object into services. When service requires the cache only for self, we are passing `IStorage`:
```
Comment (Majkl578) [inline en/caching.texy]: facing a decision
whether to pass
typo IStorage
itsels

File: en/caching.texy [hunk 3]
```diff
@@ -305,5 +305,55 @@ Framework will ensure that the body of the function will be called only by one t
...
+	private $cache;
+
+	public function __construct(Caching\Cache $cache)
+	{
+		$this->cache = $cache;
+	}
+}
+\--
+
+We are creating a cache object for every service separately:
```
Comment (Majkl578) [inline en/caching.texy]: for each

**Comment by Majkl578** [comment]
:+1:

**Comment by milo** [comment]
@Majkl578 Thanks for review.

**Pull request merged**